core/tests/performance_tests/base.py

```diff
@@ -119,3 +143,79 @@ def _test_page_load_time_for_cached_session(self):
         self.assertLessEqual(
             self.page_metrics.get_average_page_load_time_millisecs(),
             self.load_time_limit_cached_ms)
+
+    def _get_driver(self):
+        return self.data_fetcher.setup_driver()
+
+    def _check_login(self, driver):
+        driver.get(self.BASE_URL)
+        time.sleep(5)
+        resulting_url = driver.current_url
+
+        if resulting_url == self.BASE_URL + self.SPLASH_URL_SUFFIX:
+            return False
+
+        return True
+
+    def _login_user(self, driver):
+        driver.get(
+            self.BASE_URL + self.LOGIN_URL_SUFFIX + self.CONTINUE_URL_SUFFIX)
+        elem = driver.find_element_by_name("email")
+        elem.clear()
+        elem.send_keys(self.EMAIL)
+        driver.find_element_by_name('admin').click()
+        driver.find_element_by_id('submit-login').click()
+        time.sleep(5)
+        self._complete_signup(driver)
+        time.sleep(5)
+
+    def _complete_signup(self, driver):
+        driver.find_element_by_css_selector(
+            '.protractor-test-username-input').send_keys(self.USERNAME)
+        driver.find_element_by_css_selector(
+            '.protractor-test-agree-to-terms-checkbox').click()
+        driver.find_element_by_css_selector(
+            '.protractor-test-register-user').click()
+        time.sleep(5)
+
+    def _reload_explorations(self, driver):
+        driver.get(self.BASE_URL + self.ADMIN_URL_SUFFIX)
+        time.sleep(5)
+        driver.find_element_by_css_selector(
+            '.protractor-test-reload-all-explorations-button').click()
+        driver.switch_to.alert.accept()
+        time.sleep(20)
+
+    def _reload_collection(self, driver):
+        driver.get(self.BASE_URL + self.ADMIN_URL_SUFFIX)
+        time.sleep(5)
+        driver.find_element_by_css_selector(
+            '.protractor-test-reload-collection-button').click()
+        driver.switch_to.alert.accept()
+        time.sleep(5)
+
+    def _create_exploration(self, driver):
+        driver.get(self.BASE_URL + self.DASHBOARD_URL_SUFFIX)
+        driver.find_element_by_css_selector(
+            '.protractor-test-create-activity').click()
+        time.sleep(2)
+        driver.find_element_by_css_selector(
+            '.protractor-test-create-exploration').click()
+        time.sleep(5)
+        new_exploration_url = driver.current_url
+        return new_exploration_url
+
+    def _setup_login(self, driver):
+        self._login_user(driver)
+
+    def _setup_enable_explorations(self, driver):
```

Comment:
Do we ever want to load the explorations without logging in? Ditto for collections.